ui/app/routes/vault/cluster/secrets/backend/secret-edit.js

```diff
@@ -52,10 +52,11 @@ export default Ember.Route.extend(UnloadModelRoute, {
     };
     let backendModel = this.modelFor('vault.cluster.secrets.backend', backend);
     let defaultType = 'secret';
-    if (backendModel.get('type') === 'kv' && backendModel.get('options.version') === 2) {
+    let type = backendModel.get('type');
+    if ((type === 'generic' || type === 'kv') && backendModel.get('options.version') === 2) {
```

Comment:
Just a possible suggestion, you could possibly hoist this up and out as a util? Something similar to..
```
// obvs a more description name here, don't fully understand the context though, `isV2Secret` maybe?
function isSomething(model) {
  let type = model.get('type');
  return (type === 'generic' || type === 'kv') && model.get('options.version') === 2) 
}
// or more decoupled
function isSomething(type, version) {
  return (type === 'generic' || type === 'kv') && version === 2) 
}
```
You could then externalise the above and `import` it in the 3 places you are using it above, and have a simple unit test it in isolation also?